torch/csrc/jit/python/pybind_utils.cpp

```diff
@@ -69,18 +70,23 @@ IValue toIValue(py::handle obj, const TypePtr& type, c10::optional<int32_t> N) {
       size_t tuple_size = tuple.size();
       auto tuple_type = type->cast<TupleType>();
       const auto& elem_types = tuple_type->elements();
-      if (elem_types.size() != tuple_size) {
-        throw py::cast_error(c10::str(
-            "Object ",
-            py::str(obj),
-            " had a different number of elements than type ",
-            type->repr_str()));
-      }
+
       std::vector<IValue> values;
-      values.reserve(tuple_size);
+      values.reserve(elem_types.size());
       for (size_t i = 0; i < tuple_size; ++i) {
         values.push_back(toIValue(tuple[i], elem_types[i]));
       }
+
+      for (size_t i = tuple_size; i < elem_types.size(); i++) {
+        if (elem_types[i]->kind() != TypeKind::OptionalType) {
+          throw py::cast_error(c10::str(
```

Comment:
Could you elaborate what this limitation is about? what would be an example in Python that triggers this error?